### ISSUE TITLE
Slow reorg on Fractal Bitcoin

### ISSUE THREAD

**Issue by pokrovskyy** [ordinals/ord#4089]
Before you kick me for opening a ticket in main ord repo, please hear me out 😄 

I submitted an [issue in Fractal Bitcoin ord repo fork about reorg handling being beyond slow](https://github.com/fractal-bitcoin/ord/issues/17) taking 5 hrs to roll back a 2-block-deep reorg. The **reorg handling logic has NOT been changed** in the Fractal Bitcoin ord fork, so I would assume this applies to main ord too. I maintain Bitcoin index as well and frankly I haven't seen a rollback on Bitcoin for quite a while, so can't comment on how quick is it there. And even if it takes time, it's so rare and can be tolerated. On Fractal Bitcoin it seems to happen every couple hrs paralyzing the index for 5 hrs which is unacceptable.

I would assume a 2-block reorg rollback should not take 5 hrs, even if the index is 380gb, or is this expected?

**Comment by so7ow** [comment]
Did you say Fractal? 🤣

**Comment by gmart7t2** [comment]
It's slower on fractal because the index is bigger. What I do is only index a block in ord once it is 3 blocks deep in the chain. Then you almost never get affected by reorgs.

**Comment by pokrovskyy** [comment]
Makes sense @gmart7t2! Is there any config setting for such indexer lag?
